rhizome/host/bin/convert-encrypted-dek-to-vhost-backend-conf

```diff
@@ -0,0 +1,177 @@
+  end
+  unless data.key?("key2") && data["key2"].is_a?(Array) && data["key2"].length == 2 && data["key2"].all?(String)
+    raise ArgumentError, "DEK file must contain 'key2' field as an array with exactly two strings"
+  end
+end
+
+def validate_kek_file(data)
+  unless data.is_a?(Hash)
+    raise ArgumentError, "KEK file must be a dictionary"
+  end
+  data.each do |key, value|
+    unless key.is_a?(String)
+      raise ArgumentError, "KEK file keys must be strings"
+    end
+    unless value.is_a?(Hash)
+      raise ArgumentError, "KEK file values must be dictionaries"
+    end
+    required_fields = ["key", "init_vector", "auth_data", "algorithm"]
+    missing_fields = required_fields - value.keys
+    unless missing_fields.empty?
+      raise ArgumentError, "KEK entry '#{key}' is missing required fields: #{missing_fields.join(", ")}"
+    end
+    required_fields.each do |field|
+      unless value[field].is_a?(String)
+        raise ArgumentError, "KEK entry '#{key}' field '#{field}' must be a string"
+      end
+    end
+  end
+end
+
+def unwrap_key(encrypted_key, kek_data)
+  algorithm = kek_data["algorithm"]
+  fail "currently only aes-256-gcm is supported" unless algorithm == "aes-256-gcm"
+
+  decipher = OpenSSL::Cipher.new(algorithm)
+  decipher.decrypt
+  decipher.key = Base64.decode64(kek_data["key"])
+  decipher.iv = Base64.decode64(kek_data["init_vector"])
+  decipher.auth_data = kek_data["auth_data"]
+
+  auth_tag = encrypted_key[1]
+  decoded_auth_tag = Base64.decode64(auth_tag)
+
+  # We reject if auth_tag length is not within acceptable range for GCM
+  fail "Invalid auth_tag size: #{decoded_auth_tag.bytesize}" unless (12..24).cover(decoded_auth_tag.bytesize)
+
+  decipher.auth_tag = decoded_auth_tag
+  decoded_encrypted_data = Base64.decode64(encrypted_key[0])
+  decipher.update(decoded_encrypted_data) + decipher.final
+end
+
+def wrap_key(key, kek_data)
+  algorithm = kek_data["algorithm"]
+  fail "currently only aes-256-gcm is supported" unless algorithm == "aes-256-gcm"
+
+  cipher = OpenSSL::Cipher.new(algorithm)
+  cipher.encrypt
+  cipher.key = Base64.decode64(kek_data["key"])
+  cipher.iv = Base64.decode64(kek_data["init_vector"])
+  cipher.auth_data = kek_data["auth_data"]
+  [
+    cipher.update(key) + cipher.final,
+    cipher.auth_tag
+  ]
+end
+
+def wrap_key_b64(key_bytes, kek_data)
+  wrapped_key = wrap_key(key_bytes, kek_data).join
+  Base64.strict_encode64(wrapped_key).strip
+end
+
+def generate_vhost_conf(plain_keys, kek_data, vm_name)
+  config = {
+    "socket" => "/var/storage/#{vm_name}/0/vhost.sock",
+    "path" => "/var/storage/#{vm_name}/0/disk.raw",
+    "num_queues" => 1,
+    "queue_size" => 256,
+    "seg_size_max" => 65536,
+    "seg_count_max" => 4,
+    "copy_on_read" => false,
+    "poll_queue_timeout_us" => 1000,
+    "device_id" => "#{vm_name}_0",
+    "skip_sync" => false,
+    "write_through" => false
```

Comment:
To keep flush latency low, I think it's better if calculate `write_through` the same way as we do for new ubiblk volumes.